draft-sharabayko-mops-srt.md

```diff
@@ -2105,380 +2130,210 @@ with a key establishment method.
-In the UDT design, even if one peer could not decrypt, this did not mean that others should not 
-receive the information. In UDT, the key exchange is a one way process, where the key is
-regularly injected into a stream so that it can be received at any time. But with SRT the connection
-is established in the beginning, so the key exchange happens then.
-It cannot be done just anywhere in the stream.
+(e.g. AES-256) and the AES counter (which will eventually change, see {{sec-crypt-km-refresh}}).
+
+On the other side, the responder attempts to decode the wrap to obtain the Stream Encrypting Key.
+In the protocol for the wrap there is a padding, which is a known template, so the responder
+knows from the KM that it has the right KEK to decode the SEK.
+The SEK (generated and transmitted by the initiator) is random, and cannot be known in advance.
+The KEK formula is calculated on both sides, with the difference that the responder gets
+the key length (KLen) from the initiator via the key material (KM).
+It is the initiator who decides on the configured length. The responder obtains it from
+the material sent by the initiator.
+
+The responder returns the same KM message to show that it has the same information as the initiator,
+and that the encoded material will be decrypted. If the responder does not return this status,
+this means that it does not have the SEK. All incoming encrypted packets received by the responder
+will be lost (undecrypted). Even if they are transmitted
+successfully, the receiver will be unable to decrypt them, and so packets will be dropped.
+All data packets coming from responder will be unencrypted.
+
+### KM Refresh {#sec-crypt-km-refresh}
 
 The short lived SEK is regenerated for cryptographic reasons when enough packets have been
-encrypted with it by implementation of either side. One example implementation has default
-value of 2^25, with a pre-announcement period of 4000 packets (i.e. a new key is generated,
-wrapped, and sent at 2^25 minus 4000 packets). Even and odd keys are alternated this way.
-The packets with the earlier key (we will call it key #1, or "odd") will continue to be sent.
+encrypted. The KM refresh period is determined by the implementation.
+The receiver knows which SEK (odd or even) was used to encrypt the packet
+by means of the KK field of the SRT Data Packet ({{data-pkt}}).
+
+There are two variables used to determige the KM Refresh timing:
+
+- KM Refresh Period detemines the number of packets to be sent between two consecutive
+  switch points,
+
+- KM Pre-Announcement Period determines the number of packets before switch point when
+  the new key is announced., and after the switch point to decommission the old key.
+
+The recommended value of the KM Refresh Period is after 2^25 packets encrypted with the same SEK are sent.
+The recommended KM Pre-Announcement Period is 4000 packets (i.e. a new key is generated,
+wrapped, and sent at 2^25 minus 4000 packets).
```

Comment:
```suggestion
There are two variables used to determine the KM Refresh timing:

- KM Refresh Period specifies the number of packets to be sent before switching to the new SEK,

- KM Pre-Announcement Period specifies when a new key is announced a number of packets before switchover. The same value is used to determine when to decommission the old key after switchover.

The recommended KM Refresh Period is after 2^25 packets encrypted with the same SEK are sent.
The recommended KM Pre-Announcement Period is 4000 packets (i.e. a new key is generated,
wrapped, and sent at 2^25 minus 4000 packets; the old key is decommissioned at 2^25 plus 4000 packets).
```

Reply:
> KM Pre-Announcement Period specifies when a new key is announced a number of packets before switchover. The same value is used to determine when to decommission the old key after switchover.

Sounds weird... Maybe:
KM Pre-Announcement Period specifies when a new key is announced in a number of packets before key switchover. The same value is used to determine when to decommission the old key after switchover.

Reply:
Applied suggested changes with my "fix".